etna/datasets/internal_datasets.py

```diff
@@ -239,12 +257,113 @@ def get_m4_dataset(dataset_dir: Path, dataset_freq: str) -> None:
     )
 
 
+def get_m3_dataset(dataset_dir: Path, dataset_freq: str) -> None:
+    """
+    Download and save M3 dataset in different frequency modes.
+
+    The M3 dataset is a collection of 3,003 time series used for the third edition of the Makridakis forecasting
+    Competition. The M3 dataset consists of time series of yearly, quarterly, monthly and other data.
+    Each frequency mode has its own specific prediction horizon: 6 for yearly, 8 for quarterly,
+    18 for monthly, and 8 for other.
+
+    Parameters
+    ----------
+    dataset_dir:
+        The path for saving dataset locally.
+    dataset_freq:
+        Frequency mode.
+
+    References
+    ----------
+    .. [1] https://forvis.github.io/datasets/m3-data/
+    .. [2] https://forecasters.org/resources/time-series-data/m3-competition/
+    """
+    get_freq = {"monthly": "M", "quarterly": "Q-DEC", "yearly": "A-DEC", "other": "Q-DEC"}
+    get_horizon = {"monthly": 18, "quarterly": 8, "yearly": 6, "other": 8}
+    url_data = "https://forvis.github.io/data"
+    end_date = "2022-01-01"
+    freq = get_freq[dataset_freq]
+    exog_dir = dataset_dir / EXOG_SUBDIRECTORY
+
+    dataset_dir.mkdir(exist_ok=True, parents=True)
```

Comment:
I think, we can remove this line, because `exog_dir.mkdir(exist_ok=True, parents=True)` does the job.